crypt.go

```diff
@@ -454,40 +455,55 @@ func makeActivateOptions(in []string) ([]string, error) {
-	// with the fallback recovery key.
+// ActivateVolumeOptions provides options to the ActivateVolumeWith*
+// family of functions.
+type ActivateVolumeOptions struct {
+	// UserPassphraseTries specifies the maximum number of times
+	// that unsealing with a user passphrase should be attempted
+	// before failing with an error and falling back to activating
+	// with the recovery key if RecoveryKeyTries is greater than
+	// zero. Setting this to zero disables unsealing with a user
+	// passphrase - in this case, an error will be returned if the
+	// sealed key object indicates that a user passphrase has been
+	// set. With a TPM attempts to unseal with a user passphrase
+	// will stop if the TPM enters dictionary attack lockout mode
+	// before this limit is reached.
+	// It is unused by ActivateWithRecoveryKey.
+	UserPassphraseTries int
+
+	// RecoveryKeyTries specifies the maximum number of times that
+	// activation with the fallback recovery key should be
+	// attempted.  It is considered directly by
+	// ActivateWithRecoveryKey and when falling in the other
+	// cases, for example failed TPM unsealing.  Setting this to
+	// zero will disable attempts to activate with the fallback
+	// recovery key.
 	RecoveryKeyTries int
 
-	// ActivateOptions provides a mechanism to pass additional options to systemd-cryptsetup.
+	// ActivateOptions provides a mechanism to pass additional
+	// options to systemd-cryptsetup.
 	ActivateOptions []string
 
-	// LockSealedKeyAccess controls whether LockAccessToSealedKeys should be called after unsealing the TPM sealed key. It is called if
-	// this is set to true, and not called if this is set to false.
-	LockSealedKeyAccess bool
+	// LockSealedKeys controls whether access to the keys locked
+	// after unsealing. That means calling LockAccessToSealedKeys
+	// in TPM case.
+	LockSealedKeys bool
 
-	// KeyringPrefix is the prefix used for the description of any kernel keys created during activation.
+	// KeyringPrefix is the prefix used for the description of any
+	// kernel keys created during activation.
 	KeyringPrefix string
 }
 
 // ActivateVolumeWithTPMSealedKey attempts to activate the LUKS encrypted volume at sourceDevicePath and create a mapping with the
 // name volumeName, using the TPM sealed key object at the specified keyPath. This makes use of systemd-cryptsetup.
 //
-// If the TPM sealed key object has a PIN defined, then this function will use systemd-ask-password to request it. If pinReader is not
-// nil, then an attempt to read the PIN from this will be made instead by reading all characters until the first newline. The PINTries
-// field of options defines how many attempts should be made to obtain the correct PIN before failing.
+// If the TPM sealed key object has a user passphrase/PIN defined, then this function will use systemd-ask-password to request it. If passphraseReader is not
```

Comment:
That's a long comment, but I see it's consistent with the rest of the comments in the package. Nevertheless, it does not fit into a file view in a github review page, making it a bit hard to follow. Could we unify the comments style with snapd at some point?